src/param/param_slash.c

```diff
@@ -256,23 +340,96 @@ static int cmd_set(struct slash *slash) {
-	    optparse_del(parser);
-		return SLASH_EINVAL;
+	// We should iterate until we find an ending bracket ']'. Therefore we use the 'should_break' flag.
+	int should_break = 1;
+	int iterations = 0;
+	int single_value_flag = 0;
+
+	for(int i = argi; should_break == 1; i++){
+		char valuebuf[128] __attribute__((aligned(16))) = { };
+		char *arg = slash->argv[i];
+		
+		// Check if we can find a start bracket '['.
+		// If we can, then we're dealing with a value array.
+		if(strchr(arg, '[')) {
+			// If we're dealing with a value array, but only a single offset, then we should throw an error.
+			if(single_offset_flag){
+				fprintf(stderr, "cannot set array into indexed parameter value\n");
+				optparse_del(parser);
+				return SLASH_EINVAL;
+			}
+			arg++;
+			// A value array with a single element can also be passed.
+			if(arg[strlen(arg)-1] == ']'){
+				arg[strlen(arg)-1] = '\0';
+				should_break = 0;
+			}
+		}
+		// Check if we can find an ending brakcet ']'.
+		// If we can, then we should stop looping after the current iteration.
+		else if(arg[strlen(arg)-1] == ']') {
+			arg[strlen(arg)-1] = '\0';
+			should_break = 0;
+		}
+		// If no bracket was found and the current iteration is the first iteration, then we are dealing with a single value. 
+		else if(iterations == 0){
+			single_value_flag = 1;
+		}
+		
+		// Convert the valuebuf str to a value of the param type. 
+		if (param_str_to_value(param->type, arg, valuebuf) < 0) {
+			fprintf(stderr, "invalid parameter value\n");
+			optparse_del(parser);
+			return SLASH_EINVAL;
+		}
+		
+		// If we're dealing with a single value, we should loop the sliced array instead of the values.
+		// Breaking afterwards, since this means we're done setting params.
+		if(single_value_flag){
+			if(single_offset_flag){
+				param_queue_add(&queue, param, start_index, valuebuf);
+				break;
+			}
+			for(int j = start_index; j < end_index; j++){
+				param_queue_add(&queue, param, j, valuebuf);
+			}
+			break;
+		}
+		
+		// If we're not dealing with a single value, we should look at start_index.
+		// If start_index ever becomes equal to end_index, it means the amount of values are greater than the slice. 
+		if(start_index == end_index){
+			optparse_del(parser);
+			fprintf(stderr, "Values array is longer than specified slice\n");
+			return SLASH_EINVAL;
+		}
+		// Add to the queue.
+		if(param_queue_add(&queue, param, start_index++, valuebuf) < 0){
+			optparse_del(parser);
+			fprintf(stderr, "Param_queue_add failed\n");
+			return SLASH_EINVAL;
+		}
+		iterations++;
+
+		// If we hit the end value by detecting ']', and start index and end index are not equal, 
+		// then the slice length must be greater than the value array length.
+		// This shouldnt be possible, so we throw an error.
+		if(should_break == 0 && start_index != end_index){
+			optparse_del(parser);
+			fprintf(stderr, "Array slice is longer than value array\n");
+			return SLASH_EINVAL;
```

Comment:
What about the `queue`? Do you need to destroy it?

Reply:
Queue is free'd when we go out of scope. End of cmd_set.